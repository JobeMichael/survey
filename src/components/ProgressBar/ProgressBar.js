import React from "react";
import { Progress } from "antd";
import "./ProgressBar.scss";

const ProgressBar = ({ currentStep, totalQuestion, progress }) => {
  return (
    <div className="progressWrapper">
      <span className="progress-info">
        {currentStep - 1} of {totalQuestion} answered
      </span>
      <div className="progress-bar">
        <Progress percent={parseInt(progress)} />
      </div>
    </div>
  );
};

export default ProgressBar;
