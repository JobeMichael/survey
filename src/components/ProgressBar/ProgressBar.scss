@import "scss/variables";
@import "scss/mixin";

.progressWrapper {
  display: flex;
  width: 600px;
  align-items: center;
  margin: 0 auto;
  flex-direction: column;
  @include layout-lg {
    flex-direction: row;
    justify-content: center;
  }

  .ant-progress-text,
  .progress-info {
    @include text-style($weight: 700, $color: $primary-white);
  }
  .ant-progress-inner {
    border: 1px solid $primary-white;
    background-color: $primary-color;
  }
  .ant-progress-bg {
    background-color: $primary-white;
  }
  .progress-bar {
    width: 50%;
    margin-left: 12px;
  }
}
