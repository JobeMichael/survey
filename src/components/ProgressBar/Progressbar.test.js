import React from "react";
import { render } from "@testing-library/react";
import ProgressBar from "./ProgressBar";

describe("<ProgressBar/>", () => {
  it("renders the component", () => {
    const container = render(
      <ProgressBar currentStep={2} totalQuestion={10} progress={20} />
    );

    expect(container.baseElement).toMatchSnapshot();
    expect(
      container.baseElement
        .querySelectorAll("span.ant-progress-text")[0]
        .getAttribute("title")
    ).toEqual("20%");
  });
});
