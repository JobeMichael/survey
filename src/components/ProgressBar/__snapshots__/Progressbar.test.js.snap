// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar/> renders the component 1`] = `
<body>
  <div>
    <div
      class="progressWrapper"
    >
      <span
        class="progress-info"
      >
        1
         of 
        10
         answered
      </span>
      <div
        class="progress-bar"
      >
        <div
          class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
        >
          <div>
            <div
              class="ant-progress-outer"
            >
              <div
                class="ant-progress-inner"
              >
                <div
                  class="ant-progress-bg"
                  style="width: 20%; height: 8px;"
                />
              </div>
            </div>
            <span
              class="ant-progress-text"
              title="20%"
            >
              20%
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
